### PR TITLE
log specification of zero arg call was broken

### PR DESCRIPTION
**Pull request by burner**
The single arg optimization broke log calls with zero arguments. This is fixing it.
Additionally log call with a LogLevel off must not be logged.

### CODE REVIEW

**Comment by quickfur** [comment]
This doesn't have to be addressed for this PR, but next time, if you make whitespace fixes alongside other changes, please try to keep them in separate commits (or maybe even separate PRs) so that it's easier for reviewers to tell what changed in the code vs. what are just formatting adjustments.

**Comment by burner** [comment]
off course, sorry for the inconvenience

**Comment by justanotheranonymoususer** [comment]
There's also https://github.com/D-Programming-Language/phobos/pull/2976/files?w=1

File: std/experimental/logger/core.d [hunk 1]
```diff
@@ -101,6 +101,7 @@ bool isLoggingEnabled()(LogLevel ll, LogLevel loggerLL,
 
     return ll >= globalLL
```
Comment (andralex) [inline std/experimental/logger/core.d]: `ll` is a really bad name...

Comment (burner) [inline std/experimental/logger/core.d]: I know

**Comment by burner** [comment]
thanks

**Pull request merged**